test/unit/test_mc_cell_group_gpu.cpp

```diff
@@ -1,10 +1,11 @@
 #include "../gtest.h"
 
-#include <backends.hpp>
-#include <common_types.hpp>
-#include <epoch.hpp>
-#include <fvm_lowered_cell.hpp>
-#include <mc_cell_group.hpp>
+#include <arbor/common_types.hpp>
+
+#include "backends.hpp"
```

Comment:
Why are we using `"backends.hpp"` instead of `<backends.hpp>`. These files are not in the current path, so this is a bit misleading.

Reply:
The convention I'm trying to adhere to is that public headers get the angle brackets, and private ones get the quotes, just to help keep things straight. We could e.g. rename the arbor source directory back to `src`, or maybe `lib`, and then qualify the private headers with that directory prefix. In terms of confusing  header paths though, it was already a huge mess, and I think this is making it less so.

Reply:
We have discussed this in the past, and moved to using `<>` in the unit tests.

Reply:
If we see `<foo/bar.hpp>` it's not at all clear that that is referring to a private `foo/bar.hpp` or a for a `bar.hpp` belonging to a system-installed package called `foo`. I am not saying that that is a terrible thing, only that it doesn't give the clarity that you desire.

Reply:
I'd be quite happy with a `<src/arbor/foo.hpp>` or `"src/arbor/foo.hpp"` though.

Reply:
Also note that when the public/private split is complete, the number of private headers used by the tests will be very much reduced.

Reply:
And (hopefully) lastly, this distinction between <> and "" for public/private is quite common.

Reply:
How about a new macro `include_private_header(foo.hpp)`?

Joking aside, `<src/arbor/foo.hpp>` is a bit long winded, and we probably shouldn't change the directory structure of the source code just to accomodate the internal use of headers in our unit tests.

Thinking about it, the reason this makes me uncomfortable is that the quoting is being use as shorthand for "any header not part of the public API", when it is usually used as shorthand for "headers relative to this path".

Reply:
We could have grouping rules for the unit tests, e.g.

```
#include <iostream>
#include <numeric>

#include <arbor/common_types.hpp>
#include <arbor/simulation.hpp>

#include <foo.hpp>
#include <internal.hpp>
...
```

There will still be a lot of inclusion of private headers after the refactoring, because we will have tests for all the private functionality, right?

Reply:
OK, I have been doing some reading, and it would appear that using quotes for project files is a common practice.
e.g. https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

Let's argue about this in front of a black board.